compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1796,6 +1797,22 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                             );
                         }
                     }
+
+                    // UNWRAP: parsing the attribute already ensured that it had a name and integer value
+                    // that is a power of 2 and less than 2^29
+                    let (_, value) = hint.singleton_lit_list().unwrap();
```

Comment:
Suggestion:

```rs
let Some((_, value)) = hint.singleton_lit_list() else {
    bug!("parsing already checked for pow(2) and < 2^29");
};
```

Reply:
turns out this code *is* reachable after all, not sure why i didn't hit it in testing locally.  i can just ignore cases where it's malformed though, as an error will have already been emitted.